### PR TITLE
[Windows] Fix Border content position and clipping

### LINKED ISSUE #3471: Border content alignment not centered on Windows
### Description

Windows

image

Android

image

```xaml
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="ControlGallery.Pages.BorderPage"
             Title="Border"
             >
    <ContentPage.Resources>
        <Style TargetType="Border">
            <Setter Property="Stroke" Value="{StaticResource PrimaryBrush}"/>
            <Setter Property="Background" Value="Transparent"/>
            <Setter Property="HorizontalOptions" Value="Start"/>
            <Setter Property="StrokeThickness" Value="4"/>
        </Style>
    </ContentPage.Resources>
    <ContentPage.Content>
        <ScrollView>
            <VerticalStackLayout Margin="15" Spacing="8">
                <Label Text="Rectangle"
                       StyleClass="Headline" />
                <BoxView StyleClass="HRule"/>

                <Border>
                    <Border.StrokeShape>
                        <Rectangle />
                    </Border.StrokeShape>
                    <Label Text="Basic Rectangle" Margin="12,8"/>
                </Border>

                <Label Text="Round Rectangle"
                       StyleClass="Headline" />
                <BoxView StyleClass="HRule"/>

                <Border>
                    <Border.StrokeShape>
                        <RoundRectangle CornerRadius="20" />
                    </Border.StrokeShape>
                    <Label Text="Round Rectangle" Margin="12,8"/>
                </Border>

                <Label Text="Thickness"
                       StyleClass="Headline" />
                <BoxView StyleClass="HRule"/>

                <Border StrokeThickness="8">
                    <Border.StrokeShape>
                        <RoundRectangle CornerRadius="20" />
                    </Border.StrokeShape>
                    <Label Text="Round Rectangle" Margin="12,8"/>
                </Border>

                <Label Text="Stroke Dash Offset"
                       StyleClass="Headline" />
                <BoxView StyleClass="HRule"/>

                <Border StrokeDashOffset="8"
                         StrokeDashArray="1,1">
                    <Border.StrokeShape>
                        <RoundRectangle CornerRadius="20" />
                    </Border.StrokeShape>
                    <Label Text="Round Rectangle" Margin="12,8"/>
                </Border>

                <Label Text="Stroke Line Join"
                       StyleClass="Headline" />
                <BoxView StyleClass="HRule"/>

                <Border StrokeLineJoin="Bevel"
                        StrokeDashOffset="8"
                         StrokeDashArray="1,1">
                    <Border.StrokeShape>
                        <RoundRectangle CornerRadius="20" />
                    </Border.StrokeShape>
                    <Label Text="Round Rectangle" Margin="12,8"/>
                </Border>

                <Label Text="Stroke Line Cap"
                       StyleClass="Headline" />
                <BoxView StyleClass="HRule"/>

                <Border StrokeLineCap="Flat" 
                        StrokeDashOffset="8" 
                        StrokeDashArray="1,1">
                    <Border.StrokeShape>
                        <RoundRectangle CornerRadius="20" />
                    </Border.StrokeShape>
                    <Label Text="Round Rectangle" Margin="12,8"/>
                </Border>

            </VerticalStackLayout>
        </ScrollView>
    </ContentPage.Content>
        
</ContentPage>
```

Source: https://github.com/davidortinau/ControlGallery

### Steps to Reproduce

Run the code to see the border around labels
Compare Android and Windows

### Version with bug

Preview 11

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Win11

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### LINKED ISSUE #5928: [mauipalooza] Border doesn't contain or wrap image
### Description

```xaml
<Border 
        Stroke="Red"
        StrokeThickness="1" HorizontalOptions="Fill">
        <Border.StrokeShape>
            <RoundRectangle CornerRadius="7"/>
        </Border.StrokeShape>
    <Image Source="plane.png" Aspect="AspectFill" HeightRequest="145"/>
</Border>
```

I expect this to put a border all the way around the image and clip it. It doesn't. As you can see in the image, the border is behind the image which is larger.
Screen Shot 2022-04-08 at 5 38 23 PM

Simulator Screen Shot - iPhone 13 Pro - 2022-04-08 at 17 45 46

Source: https://github.com/davidortinau/dotnet-airlines

### Steps to Reproduce

Use the code provided and run it on Mac Catalyst or iOS.

### Version with bug

Release Candidate 2

### Last version that worked well

Unknown/Other

### Affected platforms

macOS

### Affected platform versions

15.4

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

This PR introduce changes in the Windows Border to:
- Fix incorrect positioning of content.
- Clip the content.

border-clip-windows

### Issues Fixed

Fixes #3471
Fixes #5928

### CODE REVIEW

File: src/Core/src/Platform/Windows/ContentPanel.cs [hunk 1]
```diff
@@ -84,6 +104,57 @@ public void UpdateBorderShape(IShape borderShape)
...
+		{
+			if (Content == null || _borderPath == null)
+				return;
+
+			var strokeThickness = _borderPath.StrokeThickness;
+
+			Content.RenderTransform = new TranslateTransform() { X = -strokeThickness, Y = -strokeThickness };
+		}
+
+		void UpdateClip(IShape? borderShape)
```
Comment (jsuarezruiz) [inline src/Core/src/Platform/Windows/ContentPanel.cs:129]: This is the main change for Clipping.

File: src/Core/src/Platform/Windows/ContentPanel.cs [hunk 2]
```diff
@@ -84,6 +104,57 @@ public void UpdateBorderShape(IShape borderShape)
...
+		}
+
+		void UpdateContent()
+		{
+			if (Content == null || _borderPath == null)
+				return;
+
+			var strokeThickness = _borderPath.StrokeThickness;
+
+			Content.RenderTransform = new TranslateTransform() { X = -strokeThickness, Y = -strokeThickness };
```
Comment (jsuarezruiz) [inline src/Core/src/Platform/Windows/ContentPanel.cs:126]: This is the main change to fix the wrong positioning.

**Comment by Redth** [comment]
@jsuarezruiz  corner radius on that sample seems broken in this PR

**Comment by jsuarezruiz** [comment]
> @jsuarezruiz corner radius on that sample seems broken in this PR

Thanks for the feedback. Could you review it again?
border-clip-windows-2

**Review by Redth — APPROVED** [review]

**Pull request merged**